### PR TITLE
Calculate serial version uid

### PR DESCRIPTION
**Pull request by sr010**
# Fixes #
Fixes: https://issues.corp.twilio.com/browse/DII-273

Testing the hash generation logic: 
image

Matches with the one generated in yoyodyne: 
image

The signatures however need to be changed to match the types in python, which is done as part of a different ticket:
image
 

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] Run `make test-docker`
- [ ] Verify affected language:
    - [ ] Generate [twilio-go](https://github.com/twilio/twilio-go) from our [OpenAPI specification](https://github.com/twilio/twilio-oai) using the [build_twilio_go.py](./examples/build_twilio_go.py) using `python examples/build_twilio_go.py path/to/twilio-oai/spec/yaml path/to/twilio-go` and inspect the diff
    - [ ] Run `make test` in `twilio-go`
    - [ ] Create a pull request in `twilio-go`
    - [ ] Provide a link below to the pull request
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-oai-generator/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please create a GitHub Issue in this repository.

### CODE REVIEW

File: src/main/java/com/twilio/oai/TwilioJavaGenerator.java [hunk 1]
```diff
@@ -183,9 +186,19 @@ public String toApiFilename(final String name) {
                         .filter(Objects::nonNull)
                         .map(this::getModel)
                         .flatMap(Optional::stream)
-                        .forEach(model -> resource.put("responseModel", model));
+                        .forEach(model -> {
+                            resource.put("responseModel", model);
```
Comment (childish-sambino) [inline src/main/java/com/twilio/oai/TwilioJavaGenerator.java:190]: Should stick this line (with the above if condition) into the new `forEach` you've added below.

File: src/main/java/com/twilio/oai/TwilioJavaGenerator.java [hunk 2]
```diff
@@ -277,6 +290,52 @@ private void addOperationName(final CodegenOperation operation, final String nam
...
+    }
+
+    private long calculate_serial_version_uid(final List<CodegenProperty> modelProperties){
+
+        String signature = calculate_signature(modelProperties);
+        String md5Hash  = getMd5(signature);
+        return Long.parseLong(getMd5(md5Hash).substring(0,12), 16);
+    }
+
+    private String calculate_signature(final List<CodegenProperty> modelProperties){
```
Comment (childish-sambino) [inline src/main/java/com/twilio/oai/TwilioJavaGenerator.java:318]: nit: Method names should be `lowerCamelCase`.

File: src/main/java/com/twilio/oai/TwilioJavaGenerator.java [hunk 3]
```diff
@@ -277,6 +290,52 @@ private void addOperationName(final CodegenOperation operation, final String nam
         operation.vendorExtensions.put("x-name", name);
         operation.vendorExtensions.put("x-name-lower", name.toLowerCase());
     }
+    private String getMd5(String input){
```
Comment (childish-sambino) [inline src/main/java/com/twilio/oai/TwilioJavaGenerator.java:293]: nit: code should read from top-to-bottom as much as reasonable. E.g., `getMd5` is only being called below so it should be defined below.

File: src/main/java/com/twilio/oai/TwilioJavaGenerator.java [hunk 4]
```diff
@@ -277,6 +290,52 @@ private void addOperationName(final CodegenOperation operation, final String nam
...
+        try {
+            MessageDigest md = MessageDigest.getInstance("MD5");
+            byte[] messageDigest = md.digest(input.getBytes());
+            BigInteger no = new BigInteger(1, messageDigest);
```
Comment (childish-sambino) [inline src/main/java/com/twilio/oai/TwilioJavaGenerator.java:297]: `no`?

File: src/main/java/com/twilio/oai/TwilioJavaGenerator.java [hunk 5]
```diff
@@ -277,6 +290,52 @@ private void addOperationName(final CodegenOperation operation, final String nam
...
+        catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private long calculate_serial_version_uid(final List<CodegenProperty> modelProperties){
+
+        String signature = calculate_signature(modelProperties);
+        String md5Hash  = getMd5(signature);
+        return Long.parseLong(getMd5(md5Hash).substring(0,12), 16);
```
Comment (childish-sambino) [inline src/main/java/com/twilio/oai/TwilioJavaGenerator.java:315]: This calls `getMd5` twice?

File: src/main/java/com/twilio/oai/TwilioJavaGenerator.java [hunk 6]
```diff
@@ -277,6 +290,52 @@ private void addOperationName(final CodegenOperation operation, final String nam
         operation.vendorExtensions.put("x-name", name);
         operation.vendorExtensions.put("x-name-lower", name.toLowerCase());
     }
+    private String getMd5(String input){
+        try {
+            MessageDigest md = MessageDigest.getInstance("MD5");
+            byte[] messageDigest = md.digest(input.getBytes());
+            BigInteger no = new BigInteger(1, messageDigest);
+            String hashtext = no.toString(16);
```
Comment (childish-sambino) [inline src/main/java/com/twilio/oai/TwilioJavaGenerator.java:298]: Seeing lots of magic numbers: 1, 16, 32. Should pull these out into named constants.

https://en.wikipedia.org/wiki/Magic_number_(programming)

File: src/main/java/com/twilio/oai/TwilioJavaGenerator.java (hunk 2 above)
Comment (sr010) [reply src/main/java/com/twilio/oai/TwilioJavaGenerator.java:318]: Fixed this

File: src/main/java/com/twilio/oai/TwilioJavaGenerator.java (hunk 1 above)
Comment (sr010) [reply src/main/java/com/twilio/oai/TwilioJavaGenerator.java:190]: Done

File: src/main/java/com/twilio/oai/TwilioJavaGenerator.java (hunk 3 above)
Comment (sr010) [reply src/main/java/com/twilio/oai/TwilioJavaGenerator.java:293]: Done

File: src/main/java/com/twilio/oai/TwilioJavaGenerator.java (hunk 4 above)
Comment (sr010) [reply src/main/java/com/twilio/oai/TwilioJavaGenerator.java:297]: Changed this

File: src/main/java/com/twilio/oai/TwilioJavaGenerator.java (hunk 6 above)
Comment (sr010) [reply src/main/java/com/twilio/oai/TwilioJavaGenerator.java]: Fixed this

File: src/main/java/com/twilio/oai/TwilioJavaGenerator.java [hunk 7]
```diff
@@ -9,11 +9,19 @@
...
 
 public class TwilioJavaGenerator extends JavaClientCodegen {
 
     // Unique string devoid of symbols.
     private static final String PATH_SEPARATOR_PLACEHOLDER = "1234567890";
+    private static final int THIRTY_TWO = 32;
+    private static final int SIXTEEN = 16;
+    private static final int ONE = 1;
+    private static final int ZERO = 0;
+    private static final int TWELVE = 12;
```
[the remark this answers is not in the document]
Comment (sr010) [reply src/main/java/com/twilio/oai/TwilioJavaGenerator.java]: Thanks for this!
Renamed them

**Pull request merged**